browser/ui/webui/ai_chat/ai_chat_untrusted_conversation_ui.cc

```diff
@@ -353,6 +355,24 @@ AIChatUntrustedConversationUI::AIChatUntrustedConversationUI(
   source->OverrideContentSecurityPolicy(
       network::mojom::CSPDirectiveName::ScriptSrc,
       "script-src 'self' chrome-untrusted://resources;");
+
+  // If the feature is not enabled then don't add the origin to the CSP.
+  if (base::FeatureList::IsEnabled(ai_chat::features::kRichSearchWidgets)) {
+    source->OverrideContentSecurityPolicy(
+        network::mojom::CSPDirectiveName::FrameSrc,
+        base::StrCat({"frame-src ",
+                      ai_chat::features::kRichSearchWidgetsOrigin.Get(),
+                      "/embed.html;"}));
+  }
+
+  // If the feature is not enabled don't specify an origin for loading the rich
+  // search widgets.
+  source->AddString(
+      "richSearchWidgetsOrigin",
+      base::FeatureList::IsEnabled(ai_chat::features::kRichSearchWidgets)
+          ? ai_chat::features::kRichSearchWidgetsOrigin.Get()
+          : "");
```

Comment:
@fallaciousreasoning @petemill  Just a reminder, please remember to add the same changes to the iOS WebUI files (`ios/browser/ai_chat/ai_chat_*`) otherwise it breaks our implementation: https://github.com/brave/brave-core/pull/32533